src/keystore.cpp

```diff
@@ -110,8 +133,10 @@ bool CBasicKeyStore::AddWatchOnly(const CScript &dest)
     LOCK(cs_KeyStore);
     setWatchOnly.insert(dest);
     CPubKey pubKey;
-    if (ExtractPubKey(dest, pubKey))
+    if (ExtractPubKey(dest, pubKey)) {
         mapWatchKeys[pubKey.GetID()] = pubKey;
+        ImplicitlyLearnRelatedKeyScripts(pubKey);
```

Comment:
In what scenario does a stale CScript result in incorrect watch-only behaviour?

Watch-only is supposed to be distinct from the normal ismine/solvability logic.

Reply:
https://github.com/bitcoin/bitcoin/pull/11403#discussion_r157382629

> Need some kind of corresponding erase in RemoveWatchOnly

@TheBlueMatt can you clarify if you still think this is a bug.
@sipa it might would be good to have a comment in RemoveWatchOnly saying CScripts are not removed but this should be harmless.

Reply:
@ryanofsky Done.